http/server_test.go

```diff
@@ -211,3 +216,95 @@ func TestNew(t *testing.T) {
 		})
 	})
 }
+
+func TestServer_LongRunningOperation(t *testing.T) {
+	DefaultWriteTimeout = 1 * time.Second
+
+	Convey("given a free port on the localhost", t, func() {
+		p, err := GetFreePort()
+		if err != nil {
+			t.Fatalf("Cannot find a free port to perform test: %v", err)
+		}
+		a := "localhost:" + strconv.Itoa(p)
+
+		Convey("with a server whose handler runs for less than the server's WriteTimout", func() {
+			h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+				time.Sleep(DefaultWriteTimeout / 100)
```

Comment:
maybe this sleep is not needed? (we are testing the case where the handle finishes fast anyway)